packages/css-modules-flow-types-loader/index.js

```diff
@@ -3,52 +3,22 @@
 import fs from 'fs';
 import printFlowDefinition from 'css-modules-flow-types-printer';
 
-/**
- * @desc
- * Parse the string representation of css-modules ICSS source code
- * and output local classNames if any were found
- *
- * @param {String} content
- * @return {Object<String, String>}
- */
 function getTokens(content) {
+  const tokens = [];
+
   // Only `locals` export is desired
   const locals = content.match(/exports\.locals = ([\s\S]*);/);
 
-  if (!locals) return {};
-
-  /**
-   * Collect object keys by parsing the string
-   * This is needed because external composes result in require calls concatenation,
-   * so we cannot simply JSON.parse (breaks on first + sign)
-   *
-   * @example
-   * // Following input css
-   * // .root {
-   * //     composes: foo from 'styles/bar.scss';
-   * //     color: red;
-   * // }
-   * // Will result in something like this pseudo-source (we get it as a string):
-   * exports.locals = {
-   *   "root": "Menu_root_1TJ" + require("-!css-loader!styles/bar.scss").locals["foo"] + "",
-   * };
-   */
-  const tokens = [];
+  if (!locals) return tokens;
   let match;
 
-  // RegExp.exec is state-full, so we need to
-  // initialize new one for each run
+  // RegExp.exec is state-full, so we need to initialize new one for each run
   const re = /"(.*?)":.*\n/g;
   while ((match = re.exec(locals[1])) !== null) {
     if (match[1]) tokens.push(match[1]);
```

Comment:
Is there ever a case where `match[1]` is false?

Reply:
Theoretically it could be an empty string, should the key be empty…

It shouldn't really happen it the input works correctly being a transformation of stylesheet to to an JS object because class name cannot be empty.
It is however possible for js object  (which is our direct data input, be it as a string) to be an empty string

![selection_003](https://user-images.githubusercontent.com/11244314/31543743-debd579c-b017-11e7-83f1-015ccbf276cd.png)

Reply:
However if we would print the resulting flow key as:
```javascript
type x = {|
    +"": string,
|}
```

it should be apparently OK as far as flow is concerned ... so mabye this is extraneous in the end…

Reply:
Thanks for investigating this